File: app/src/main/java/ch/berta/fabio/tipee/ui/dialogs/TippingNotCommonDialogFragment.java
package ch.berta.fabio.tipee.ui.dialogs;

import android.support.v7.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

import ch.berta.fabio.tipee.R;

/**
 * A {@link android.app.DialogFragment} subclass that displays a dialog informing the user that
 * tipping is not common in the country selected by the user.
 *
 * @author Fabio Berta
 */
public class TippingNotCommonDialogFragment extends DialogFragment {
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
        dialogBuilder.setTitle(R.string.tipping_not_common_dialog_title)
                .setMessage(R.string.tipping_not_common_dialog_message)
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dismiss();
                    }
                });
        return dialogBuilder.create();
    }
}

File: app/src/main/java/ch/berta/fabio/tipee/features/tip/dialogs/TippingNotCommonDialogFragment.kt
package ch.berta.fabio.tipee.features.tip.dialogs

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v4.app.FragmentManager
import android.support.v7.app.AlertDialog
import ch.berta.fabio.tipee.R

/**
 * A [android.app.DialogFragment] subclass that displays a dialog informing the user that
 * tipping is not common in the selectedCountry selected by the user.

 * @author Fabio Berta
 */
class TippingNotCommonDialogFragment : DialogFragment() {

    companion object {
        val tag: String = TippingNotCommonDialogFragment::class.java.canonicalName

        fun display(fm: FragmentManager) {
            val dialog = TippingNotCommonDialogFragment()
            dialog.show(fm, tag)
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialogBuilder = AlertDialog.Builder(activity)
        dialogBuilder.setTitle(R.string.dialog_tipping_not_common_title)
                .setMessage(R.string.dialog_tipping_not_common_message)
                .setPositiveButton(android.R.string.ok) { _, _ -> dismiss() }
        return dialogBuilder.create()
    }
}